powershell_completions.go

```diff
@@ -11,90 +8,253 @@ import (
+        $comp = $_
+
+
+        # Powershell supports three different completion modes
+        # - TabCompleteNext (default windows style - on each key press the next option is displayed)
+        # - Complete (works like bash)
+        # - MenuComplete (works like zsh)
+        # You set the mode with Set-PSReadLineKeyHandler -Key Tab -Function <mode>
+
+        # CompletionResult Arguments:
+        # 1) CompletionText text to be used as the auto completion result
+        # 2) ListItemText   text to be displayed in the suggestion list
+        # 3) ResultType     type of completion result
+        # 4) ToolTip        text for the tooltip with details about the object
+
+        switch ($Mode) {
+
+            # bash like
+            "Complete" {
+
+                if ($Values.Length -eq 1) {
+                    __%[1]s_debug "Only one completion left"
+
+                    # insert space after value
+                    [System.Management.Automation.CompletionResult]::new("$($comp.Name)$Space", "$($comp.Name)", 'ParameterValue', "$($comp.Description)")
+
+                } else {
+                    ##TODO Trim Description if to long
+
+                    # Add the proper number of spaces to align the descriptions
+                    while($comp.Name.Length -lt $Longest) {
+                        $comp.Name = $comp.Name + " "
+                    }
+
+                    [System.Management.Automation.CompletionResult]::new("$($comp.Name)  ($($comp.Description))", "$($comp.Name)  ($($comp.Description))", 'ParameterValue', "$($comp.Description)")
+                }
+             }
+
+            # zsh like
+            "MenuComplete" {
+                # insert space after value
+                # MenuComplete will automatically show the ToolTip of 
+                # the highlighted value at the bottom of the suggestions.
+                [System.Management.Automation.CompletionResult]::new("$($comp.Name)$Space", "$($comp.Name)", 'ParameterValue', "$($comp.Description)")
+            }
+
+            # TabCompleteNext and in case we get something unknown
+            Default {
+                # Like MenuComplete but we dont want to add a space here because 
+                # the user need to press space anyway to get the completion.
+                # Description will not be shown because thats not possible with TabCompleteNext
+                [System.Management.Automation.CompletionResult]::new("$($comp.Name)", "$($comp.Name)", 'ParameterValue', "$($comp.Description)")
+            }
+        }
+
+    }
+}
+`, name, compCmd,
+		ShellCompDirectiveError, ShellCompDirectiveNoSpace, ShellCompDirectiveNoFileComp,
+		ShellCompDirectiveFilterFileExt, ShellCompDirectiveFilterDirs))
 }
 
 // GenPowerShellCompletion generates PowerShell completion file and writes to the passed writer.
-func (c *Command) GenPowerShellCompletion(w io.Writer) error {
+func (c *Command) GenPowerShellCompletion(w io.Writer, includeDesc bool) error {
```

Comment:
This change is not backwards compatible.  We'll need to add new functions instead.

Reply:
Yes I'm aware. I'm waiting for #1146 to decide how we want to handle this.

Reply:
Since we know this change cannot be done, I suggest you implement whatever API you prefer and restore the previous methods.  This will at least put the PR in a state that can be merged.